### DISCUSSION TITLE
Add UI Scaling Ability

### DISCUSSION THREAD

**Discussion by jrodden1** [andrewdavidmackenzie/pigg#746] [Ideas]
## Description
Often times, Pi's are running on tiny little 7" touchscreens at 800x600 or 1024x600.  Since I was unable to find a UI scale option (and `Ctrl` + `-` and `Ctrl` + `+` had no effect), it is rather cumbersome to navigate.  It looks super nice but would be wonderful to be able to scale it down so that it'll all fit on a 800x600 screen.  I have attached a screenshot of what it looks like currently on the little 7" touchscreen running 800x600 to illustrate the issue.  

## Desired Results 
* Have the ability to set various UI scaling levels 
* And/or have an option for 'Small Screens' that is designed for smaller resolution screens. 
* And/or have the ability to cherry pick which pins I see and monitor so that it fits on the tiny screen.  

## Current Results 
* Unable to change the UI scaling
* Having to use `Alt` + `Click and Drag` to navigate around the window to resize it and having to deal with scroll bars and big UI elements that don't scale. 
![gui on 800x600](https://github.com/user-attachments/assets/f04a1f5e-b411-4238-91eb-60f350ed40c2)

## Comments 
This app looks to be SUPER helpful, by the way.  I just figured it would be helpful to have a scaled version that fits on tiny screens too. :) Thanks for your work @andrewdavidmackenzie!

**Comment by sundaram123krishnan** [comment]
We can manipulate the `scale_factor` and listen for keyboard events `Ctrl` + `+` and `Ctrl` + `-`
@andrewdavidmackenzie

**Comment by andrewdavidmackenzie** [comment]
Yes, we've been concerned about window size and moved some things to gain space, but it's still big.

I'll look into scaling. Font sizes might be an issue, but we'll see 

Meanwhile, you might try the other layout. Then maybe less horizontal scrolling that way...?

**Comment by andrewdavidmackenzie** [comment]
Ability to view/hide pins is also interesting. I might break that out into a separate GH issue.

**Comment by andrewdavidmackenzie** [comment]
Will explore via https://github.com/andrewdavidmackenzie/pigg/issues/747, thanks for the feedback and writing it up so well.

**Comment by jrodden1** [comment]
You guys are great!  

Also another minor note, the 7" touch screen I have is 800 x **480** actually (instead of 800x600) so height is very limited.

**Comment by andrewdavidmackenzie** [comment]
Ok!

Touchscreen eh!?

What about size of controls for touch?
Is that a separate issue, or (physical) size on screen is an issue.

**Comment by andrewdavidmackenzie** [comment]
Thought better of it, and moved my comments to the issue:
https://github.com/andrewdavidmackenzie/pigg/issues/747

Will close this conversation in favor of the issue if that's OK

**Comment by andrewdavidmackenzie** [comment]
@jrodden1 independently of scale factor changes, I have created a new "compact" view, that only shows pins configured, and moves the drop-down list (Input, Putput, None) to a menu on the pin, saving considerable height and width in the compact view, and width in the BCM and Board Pin layouts.

You can see some screen shots here in the PR: https://github.com/andrewdavidmackenzie/pigg/pull/836

WDYT?
